debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -102,16 +113,58 @@ public final void start(Map<String, String> props) {
 
     @Override
     public final List<SourceRecord> poll() throws InterruptedException {
-        return doPoll();
+        boolean started = startIfNeededAndPossible();
+
+        // in backoff period after a retriable exception
+        if (!started) {
+            return Collections.emptyList();
+        }
+
+        try {
+            return doPoll();
+        }
+        catch (RetriableException e) {
```

Comment:
Am I missing something here since the doPoll() method below throws InterruptedException but not RetriableException?

Reply:
`RetriableException` is a runtime exception, so it doesn't have to be declared explicitly in method sigs.